pyproject.toml

```diff
@@ -5,7 +5,7 @@ requires = [
   # https://numpy.org/devdocs/user/depending_on_numpy.html#build-time-dependency
   # from https://github.com/scipy/oldest-supported-numpy/pull/78#issuecomment-1747936818:
   "oldest-supported-numpy; platform_python_implementation != 'PyPy'",
-  "numpy < 2.0.0; platform_python_implementation=='PyPy'",
+  "numpy; platform_python_implementation=='PyPy'",
```

Comment:
You'll need to drop `oldest-supported-numpy`, and just use `numpy>=2.0.0rc1` instead of either. See the [information for downstream package authors](https://numpy.org/doc/stable/dev/depending_on_numpy.html#numpy-2-0-specific-advice).